content/flux/v0.x/get-started/syntax-basics.md

```diff
@@ -0,0 +1,353 @@
+  - [Regular expression types](#regular-expression-types)
+- [Packages](#packages)
+- [Examples of basic syntax](#examples-of-basic-syntax)
+  - [Define data stream variables](#define-data-stream-variables)
+  - [Define custom functions](#define-custom-functions)
+
+## Pipe-forward operator
+The **pipe-forward operator** (`|>`) sends the output of one function as input to the next function.
+In the [water treatment metaphor](/flux/v0.x/get-started/#flux-overview),
+the pipe-forward operator is the pipe that carries water (or data) through the pipeline.
+
+```js
+data
+  |> someFunction()
+  |> anotherFunction()
+```
+
+## Simple expressions
+Flux supports basic expressions.
+For example:
+
+```js
+1 + 1
+// Returns 2
+
+10 * 3
+// Returns 30
+
+(12.0 + 18.0) / (2.0 ^ 2.0) + (240.0 % 55.0)
+// Returns 27.5
+
+"John " + "Doe " + "is here!"
+// Returns John Doe is here!
+```
+
+_For information about operator precedence, see
+[Flux Operators – Operator precedence](/flux/v0.x/spec/operators/#operator-precedence)._
+
+## Predicate expressions
+A predicate is an expression that returns `true` or `false`.
+A predicate expression consists of [comparison operators](/flux/v0.x/spec/operators/#comparison-operators)
+and [logical operators](/flux/v0.x/spec/operators/#logical-operators) that specify a
+relationship between values or operands.
+For example:
+
+```js
+"John" == "John"
+// Returns true
+
+41 < 30
+// Returns false
+
+"John" == "John" and 41 < 30
+// Returns false
+
+"John" == "John" or 41 < 30
+// Returns true
+```
+
+Flux uses predicate expressions when [filtering data](/flux/v0.x/get-started/query-basics/#filter)
+or in [conditional expressions](/flux/v0.x/spec/expressions/#conditional-expressions).
+
+## Variables
+Assign an expression to a variable using the
+[assignment operator (`=`)](/flux/v0.x/spec/operators/#assignment-operators).
+Use the name (identifier) of a variable to return its value:
+
+```js
+s = "foo" // string
+i = 1 // integer
+f = 2.0 // float (floating point number)
+
+s // Returns foo
+i // Returns 1
+f // Returns 2
+```
+
+Variables can be assigned to any [Flux data type](/flux/v0.x/data-types/).
+
+## Data types
+Flux supports many data types grouped into the following categories:
+
+- [Basic types](#basic-types)
+- [Composite types](#composite-types)
+- [Regular expression types](#regular-expression-types)
+
+### Basic types
+The following basic types can be represented with literal values:
+
+- [Boolean](/flux/v0.x/data-types/basic/bool/)
+- [Duration](/flux/v0.x/data-types/basic/duration/)
+- [String](/flux/v0.x/data-types/basic/string/)
+- [Time](/flux/v0.x/data-types/basic/time/)
+- [Float](/flux/v0.x/data-types/basic/float/)
+- [Integer](/flux/v0.x/data-types/basic/integer/)
+
+```js
+// Boolean
+true
+
+// Duration
+23h4m5s
+
+// String
+"foo"
+
+// Time
+2021-01-01T00:00:00Z
+
+// Float
+1.0
+
+// Integer
+1
+```
+
+The following basic types do not have a literal syntax, but can be created in other ways.
```

Comment:
how many/what other ways?

Reply:
```suggestion
The following basic types do not have a literal syntax, but can be created in other ways:
```

Reply:
would it make sense to say 
The following basic types do not have a literal syntax:
...
Therefore, you must do the following to create...: